test/machines/machines_test.go

```diff
@@ -340,6 +341,36 @@ var _ = framework.SigKubeDescribe("Machines", func() {
 			f.DeleteMachineAndWait(masterMachine, acw)
 		})
 
-	})
+		It("Can not delete instances from type master", func() {
+			machineProviderConfig, err := utils.TestingMachineProviderConfig(awsCredSecret.Name, cluster.Name)
+			Expect(err).NotTo(HaveOccurred())
+			masterMachine := manifests.MasterMachine(cluster.Name, cluster.Namespace, machineProviderConfig)
+			masterMachine.ObjectMeta.Labels["sigs.k8s.io/cluster-api-machine-type"] = "master"
+			f.CreateMachineAndWait(masterMachine, acw)
+			machinesToDelete.AddMachine(masterMachine, f, acw)
+
+			By(fmt.Sprintf("Deleting %q machine", masterMachine.Name), func() {
+				err := f.CAPIClient.ClusterV1alpha1().Machines(masterMachine.Namespace).Delete(masterMachine.Name, &metav1.DeleteOptions{})
+				f.IgnoreNotFoundErr(err)
+			})
+
+			// Verify the testing machine has not been destroyed
+			By("Verify instance is not terminated", func() {
+				err := wait.Poll(framework.PollInterval, framework.PoolTimeout, func() (bool, error) {
+					_, err := f.CAPIClient.ClusterV1alpha1().Machines(masterMachine.Namespace).Get(masterMachine.Name, metav1.GetOptions{})
+					if err == nil {
+						log.Info("Waiting for machine to not be deleted")
```

Comment:
`"Waiting for machine to not be deleted"` does not make sense if you do `"Verify instance is not terminated"`. Maybe a copy-paste error?